x-pack/plugins/security_solution/public/entity_analytics/hooks/use_risk_contributing_alerts.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useQueryAlerts } from '../../detections/containers/detection_engine/alerts/use_query';
+import { ALERTS_QUERY_NAMES } from '../../detections/containers/detection_engine/alerts/constants';
+
+import type {
+  UserRiskScore,
+  HostRiskScore,
+} from '../../../common/search_strategy/security_solution/risk_score/all';
+import {
+  isUserRiskScore,
+  RiskScoreFields,
+} from '../../../common/search_strategy/security_solution/risk_score/all';
+import { getStartDateFromRiskScore } from '../common/get_start_date_from_risk_score';
+import { useRiskEngineSettings } from '../api/hooks/use_risk_engine_settings';
+
+interface UseRiskContributingAlerts {
+  riskScore: UserRiskScore | HostRiskScore | undefined;
+  fields?: string[];
+}
+
+interface Hit {
+  fields: Record<string, string[]>;
+  _index: string;
+  _id: string;
+}
+
+interface UseRiskContributingAlertsResult {
+  loading: boolean;
+  error: boolean;
+  data?: Hit[];
+}
+
+const ALERTS_SIZE = 1000;
+
+/**
+ * Fetches alerts related to the risk score
+ */
+export const useRiskContributingAlerts = ({
+  riskScore,
+  fields,
+}: UseRiskContributingAlerts): UseRiskContributingAlertsResult => {
+  const { data: riskEngineSettings } = useRiskEngineSettings();
+
+  const { loading, data, setQuery } = useQueryAlerts<Hit, unknown>({
+    // is empty query, to skip fetching alert, until we have risk engine settings
+    query: {},
+    queryName: ALERTS_QUERY_NAMES.BY_ID,
+  });
+
+  useEffect(() => {
+    if (!riskEngineSettings?.range?.start || !riskScore) return;
+
+    let entityField: string;
```

Comment:
opinion: maybe this code (L60-69) could be moved into `getQuery` ?

Reply:
Sure, i also moved getQuery in separate file, and added tests